tchannel-core/src/main/java/com/uber/tchannel/channels/Connection.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2015 Uber Technologies, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.tchannel.channels;
+import com.uber.tchannel.messages.InitMessage;
+import io.netty.channel.Channel;
+
+import java.util.Map;
+
+/**
+ * Connection represents a connection to a remote address
+ */
+public class Connection {
+    public Channel channel;
+    public String remoteAddress = null;
+    public Direction direction = Direction.NONE;
```

Comment:
Direction matters when talking to Hyperbahn, since affinity connections are always IN. We want all clients to use affinity connections to talk to Hyperbahn.

The current code also has a bug that leaks connections when it has an OUT connection to the host and later has an IN connection back from the same host.